src/foam/parse/QueryParser.java

```diff
@@ -394,6 +394,8 @@ private Grammar getGrammar() {
       new LiteralIC("me"),
       new foam.lib.parse.Not(grammar.sym("CHAR"))
     ));
+    // FIXME: this will never work as x is an empty parser context,
+    // event  "X", set by the Parser setup is also empty.
```

Comment:
Normally the Context X is stored in the ParserContext.

Reply:
Perhaps this is the intent, but it's not working that way. 
Inside the parser, there is no access to any context which is not empty.